standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -10330,6 +10330,108 @@ public boolean deletePartitionColumnStatistics(String catName, String dbName, St
+      } else {
+        filter = "partition.partitionName == t1 && partition.table.database.name == t2 && partition.table.tableName == t3 && partition.table.database.catalogName == t4" + (engine != null ? " && engine == t5" : "");
+        parameters = "java.lang.String t1, java.lang.String t2, java.lang.String t3, java.lang.String t4" + (engine != null ? ", java.lang.String t5" : "");
+      }
+      query.setFilter(filter);
+      query.declareParameters(parameters);
+      if (colNames != null) {
+        query.setUnique(true);
+        for (String colName : colNames){
+          // trim the extra spaces, and change to lowercase
+          normalizeIdentifier(colName);
+        }
+        if (engine != null) {
+          mStatsObj =
+                  (MPartitionColumnStatistics) query.executeWithArray(partName.trim(),
+                          normalizeIdentifier(dbName),
+                          normalizeIdentifier(tableName),
+                          colNames,
+                          normalizeIdentifier(catName),
+                          engine);
+        } else {
+          mStatsObj =
+                  (MPartitionColumnStatistics) query.executeWithArray(partName.trim(),
+                          normalizeIdentifier(dbName),
+                          normalizeIdentifier(tableName),
+                          colNames,
+                          normalizeIdentifier(catName));
+        }
+        pm.retrieve(mStatsObj);
+        if (mStatsObj != null) {
+          pm.deletePersistent(mStatsObj);
+        } else {
+          throw new NoSuchObjectException("Column stats doesn't exist for table="
+                  + TableName.getQualified(catName, dbName, tableName) +
+                  " partition=" + partName + " col=" + String.join(", ", colNames));
+        }
+      } else {
+        if (engine != null) {
+          mStatsObjColl =
+                  (List<MPartitionColumnStatistics>) query.executeWithArray(partName.trim(),
+                          normalizeIdentifier(dbName),
+                          normalizeIdentifier(tableName),
+                          normalizeIdentifier(catName),
+                          engine);
+        } else {
+          mStatsObjColl =
+                  (List<MPartitionColumnStatistics>) query.executeWithArray(partName.trim(),
+                          normalizeIdentifier(dbName),
+                          normalizeIdentifier(tableName),
+                          normalizeIdentifier(catName));
+        }
+        pm.retrieveAll(mStatsObjColl);
+        if (mStatsObjColl != null) {
+          pm.deletePersistentAll(mStatsObjColl);
+        } else {
+          throw new NoSuchObjectException("Column stats don't exist for table="
+                  + TableName.getQualified(catName, dbName, tableName) + " partition" + partName);
+        }
+      }
+      ret = commitTransaction();
```

Comment:
after deleting the column stats, should we mark the `COLUMN_STATS_ACCURATE` to false in partition's parameters?

Reply:
That's a good point. I don't have an answer for it now. 
We had the feature of dropping column stats for single column before, and the `COLUMN_STATS_ACCURATE ` was not mark to false.
So let's discuss about that later.